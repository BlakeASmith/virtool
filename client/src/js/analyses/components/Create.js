import React from "react";
import { push } from "connected-react-router";
import { connect } from "react-redux";
import { get, map, find, forEach, xorBy } from "lodash-es";
import { Modal, ListGroup, Col, Label } from "react-bootstrap";
import { AlgorithmSelect, Button, ListGroupItem, NoneFound, Checkbox } from "../../base/index";
import { getTaskDisplayName, routerLocationHasState } from "../../utils/utils";
import { analyze } from "../actions";

export const IndexSelect = ({ indexes, onSelect, selected, error }) => {
    const errorStyle = error.length ? { border: "1px solid #d44b40" } : { border: "1px solid transparent" };

    return (
        <div style={{ marginBottom: "16px" }}>
            <label className="control-label">References</label>
            {indexes.length ? (
                <div>
                    <ListGroup
                        style={{
                            maxHeight: "165px",
                            overflowY: "auto",
                            marginBottom: "3px",
                            ...errorStyle
                        }}
                    >
                        {map(indexes, index => (
                            <ListGroupItem
                                key={index.id}
                                onClick={() => onSelect({ id: index.id, refId: index.reference.id })}
                            >
                                <Col xs={1}>
                                    <Checkbox checked={!!find(selected, ["id", index.id])} />
                                </Col>
                                <Col xs={8}>
                                    <strong>{index.reference.name}</strong>
                                </Col>
                                <Col xs={3}>
                                    Index Version <Label>{index.version}</Label>
                                </Col>
                            </ListGroupItem>
                        ))}
                    </ListGroup>
                    <Col xs={12}>
                        <div className="input-form-error">
                            <span className="input-error-message">{error}</span>
                        </div>
                    </Col>
                </div>
            ) : (
                <NoneFound noun="source references" />
            )}
        </div>
    );
};

const MultiSummary = ({ algorithm, samples, selected }) => {
    let prefix;
    let suffix;

    if (selected.length) {
        prefix = `Start ${selected.length} ${getTaskDisplayName(algorithm)} ${selected.length > 1 ? "jobs" : "job"}`;

        if (samples) {
            suffix = ` ${this.props.samples.length} ${this.props.samples.length > 1 ? "samples" : "sample"}.`;
        }
    }

    return (
        <div style={{ float: "left" }}>
            {prefix}
            {suffix}
        </div>
    );
};

const getInitialState = () => ({
    algorithm: "pathoscope_bowtie",
    selected: [],
    error: ""
});

export class CreateAnalysis extends React.Component {
    constructor(props) {
        super(props);
        this.state = getInitialState();
    }

    handleSelectIndex = index => {
        this.setState({ selected: xorBy(this.state.selected, [index], "id"), error: "" });
    };

    handleSubmit = e => {
        e.preventDefault();

        if (!this.state.selected.length) {
            return this.setState({ error: "Please select reference(s)" });
        }

        this.props.onAnalyze(this.props.id, this.state.selected, this.state.algorithm, this.props.userId);
        this.props.onHide();
    };

    render() {
        const { selected, algorithm } = this.state;

        return (
            <Modal show={this.props.show} onHide={this.props.onHide} onExited={() => this.setState(getInitialState())}>
                <Modal.Header>New Analysis</Modal.Header>
                <form onSubmit={this.handleSubmit}>
                    <Modal.Body>
                        <AlgorithmSelect
                            value={algorithm}
                            onChange={e => this.setState({ algorithm: e.target.value })}
                            hasHmm={this.props.hasHmm}
                        />
                        <IndexSelect
                            indexes={this.props.indexes}
                            onSelect={this.handleSelectIndex}
                            selected={selected}
                            error={this.state.error}
                        />
                    </Modal.Body>
                    <Modal.Footer>
                        <MultiSummary
                            algorithm={algorithm}
                            samples={this.props.samples}
                            selected={this.state.selected}
                        />
                        <Button type="submit" bsStyle="primary" icon="play" disabled={!!this.state.error}>
                            Start
                        </Button>
                    </Modal.Footer>
                </form>
            </Modal>
        );
    }
}

const mapStateToProps = state => ({
    show: routerLocationHasState(state, "createAnalysis"),
    sampleId: get(state, "samples.detail.id"),
    indexes: state.analyses.readyIndexes
});

const mapDispatchToProps = dispatch => ({
    onAnalyze: (sampleId, references, algorithm, userId) => {
        forEach(references, entry => dispatch(analyze(sampleId, entry.refId, algorithm, userId)));
    },
    onHide: () => {
        dispatch(push({ ...window.location, state: { createAnalysis: false } }));
    }
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(CreateAnalysis);
