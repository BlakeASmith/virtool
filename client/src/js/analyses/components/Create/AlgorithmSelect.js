import React from "react";
import PropTypes from "prop-types";
import { map } from "lodash-es";

import { getTaskDisplayName } from "../../../utils/utils";
import { Input } from "../../../base";

export const getCompatibleAlgorithms = libraryType => {
    if (libraryType === "amplicon") {
        return ["aodp"];
    }

    return ["pathoscope_bowtie", "nuvs"];
};

/**
 * An input-based component for selecting an algorithm (eg. pathoscope_bowtie), by its display name
 * (eg. Pathoscope Bowtie).
 *
 */
export const AlgorithmSelect = ({ hasHmm, libraryType, value, onChange }) => (
    <Input name="algorithm" type="select" label="Algorithm" value={value} onChange={onChange}>
        {map(getCompatibleAlgorithms(libraryType), algorithm => (
            <option key={algorithm} value={algorithm} disabled={algorithm === "nuvs" && !hasHmm}>
                {getTaskDisplayName(algorithm)}
            </option>
        ))}
    </Input>
);

AlgorithmSelect.propTypes = {
    libraryType: PropTypes.string.isRequired,
    value: PropTypes.oneOf(["pathoscope_bowtie", "nuvs", "aodp"]),
    onChange: PropTypes.func,
    hasHmm: PropTypes.bool
};
