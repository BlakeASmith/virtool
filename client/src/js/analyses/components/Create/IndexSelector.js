import { map, some } from "lodash-es";
import React, { useCallback } from "react";
import { Label, ListGroup } from "react-bootstrap";
import styled from "styled-components";
import { Checkbox, ListGroupItem, NoneFound } from "../../../base";

const StyledIndexSelectItem = styled(ListGroupItem)`
    align-items: center;
    display: flex;
    justify-content: space-between;

    .label {
        margin-left: 3px;
    }

    span {
        align-items: center;
        display: flex;
    }

    strong {
        margin-left: 8px;
    }
`;

export const IndexSelectItem = ({ id, reference, isSelected, version, onSelect }) => {
    const handleClick = useCallback(() => onSelect({ id, refId: reference.id }), []);
    return (
        <StyledIndexSelectItem onClick={handleClick}>
            <span>
                <Checkbox checked={isSelected} />
                <strong>{reference.name}</strong>
            </span>
            <span>
                Index Version <Label>{version}</Label>
            </span>
        </StyledIndexSelectItem>
    );
};

const StyledIndexSelectList = styled(ListGroup)`
    border: 1px solid ${props => (props.error.length ? "#d44b40" : "transparent")};
    max-height: 165px;
    overflow-y: auto;
    margin-bottom: 3px;
`;

const IndexSelectList = ({ error, indexes, selected, onSelect }) => {
    const indexComponents = map(indexes, index => {
        const isSelected = some(selected, { id: index.id });
        return <IndexSelectItem key={index.id} {...index} isSelected={isSelected} onSelect={onSelect} />;
    });

    return <StyledIndexSelectList error={error}>{indexComponents}</StyledIndexSelectList>;
};

export const IndexSelector = ({ indexes, onSelect, selected, error }) => {
    return (
        <div style={{ marginBottom: "16px" }}>
            <label className="control-label">References</label>
            {indexes.length ? (
                <IndexSelectList error={error} indexes={indexes} selected={selected} onSelect={onSelect} />
            ) : (
                <NoneFound noun="source references" />
            )}
        </div>
    );
};
