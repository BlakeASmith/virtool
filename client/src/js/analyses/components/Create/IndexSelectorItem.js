import React, { useCallback } from "react";
import styled from "styled-components";
import { Checkbox, BoxGroupSection, Label } from "../../../base";

const StyledIndexSelectorItem = styled(BoxGroupSection)`
    align-items: center;
    display: flex;
    user-select: none;

    > span:last-child {
        margin-left: auto;
    }
`;

export const IndexSelectorItem = ({ id, reference, isSelected, version, onSelect }) => {
    const handleClick = useCallback(() => onSelect({ id, refId: reference.id }), [id, reference]);

    return (
        <StyledIndexSelectorItem active={isSelected} onClick={handleClick}>
            <Checkbox checked={isSelected} />
            <strong>{reference.name}</strong>
            <span>
                Index Version <Label>{version}</Label>
            </span>
        </StyledIndexSelectorItem>
    );
};
