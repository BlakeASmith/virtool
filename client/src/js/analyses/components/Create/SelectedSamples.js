import React from "react";
import styled from "styled-components";
import { map } from "lodash-es";

import { Badge, ListGroup } from "react-bootstrap";
import { ListGroupItem } from "../../../base";

export const SelectedSamplesLabel = ({ count }) => {
    let tail;

    if (count > 1) {
        tail = (
            <React.Fragment>
                s <Badge>{count}</Badge>
            </React.Fragment>
        );
    }

    return <label className="control-label">Sample{tail}</label>;
};

const SelectedSamplesListGroupContainer = styled(ListGroup)`
    margin-bottom: 16px;
    max-height: 220px;
    overflow-y: ${props => (props.count > 1 ? "scroll" : "auto")};

    div {
        margin-bottom: 0;
    }
`;

export const SelectedSamples = ({ samples }) => {
    const count = samples.length;

    const sampleComponents = map(samples, ({ id, name }) => (
        <ListGroupItem key={id} disabled>
            {name}
        </ListGroupItem>
    ));

    return (
        <React.Fragment>
            <SelectedSamplesLabel count={count} />
            <SelectedSamplesListGroupContainer count={count}>
                <ListGroup>{sampleComponents}</ListGroup>
            </SelectedSamplesListGroupContainer>
        </React.Fragment>
    );
};
