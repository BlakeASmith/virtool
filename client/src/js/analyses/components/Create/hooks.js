import { xor, xorBy } from "lodash-es";
import { useEffect, useState } from "react";

export const useCreateAnalysis = (dataType, defaultSubtraction) => {
    const [error, setError] = useState("");
    const [indexes, setIndexes] = useState([]);
    const [subtraction, setSubtraction] = useState(defaultSubtraction);
    const [workflows, setWorkflows] = useState([]);

    const toggleIndex = index => {
        setIndexes(prev => xorBy(prev, [index], "id"));
        setError("");
    };

    const toggleWorkflow = workflow => {
        setWorkflows(xor(workflows, [workflow]));
    };

    useEffect(() => {
        setIndexes([]);
        setWorkflows([]);
    }, [dataType]);

    return {
        error,
        indexes,
        subtraction,
        workflows,
        setError,
        setSubtraction,
        toggleIndex,
        toggleWorkflow
    };
};
