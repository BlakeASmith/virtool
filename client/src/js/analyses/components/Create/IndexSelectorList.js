import { map, some } from "lodash-es";
import React from "react";
import styled from "styled-components";
import { NoneFound, BoxGroup } from "../../../base";
import { IndexSelectorItem } from "./IndexSelectorItem";

const StyledIndexSelectorList = styled(BoxGroup)`
    border: 1px solid ${props => (props.error.length ? "#d44b40" : "transparent")};
    max-height: 165px;
    overflow-y: auto;
    margin-bottom: 3px;
`;

export const IndexSelectorList = ({ error, indexes, selected, onSelect }) => {
    if (indexes.length) {
        const indexComponents = map(indexes, index => {
            const isSelected = some(selected, { id: index.id });
            return <IndexSelectorItem key={index.id} {...index} isSelected={isSelected} onSelect={onSelect} />;
        });

        return <StyledIndexSelectorList error={error}>{indexComponents}</StyledIndexSelectorList>;
    }

    return <NoneFound />;
};
