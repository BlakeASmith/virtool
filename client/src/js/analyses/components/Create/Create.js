import { forEach, xorBy } from "lodash-es";
import React from "react";
import { Modal } from "react-bootstrap";
import { connect } from "react-redux";
import { pushState } from "../../../app/actions";
import { AlgorithmSelect, Button, Flex, FlexItem } from "../../../base";
import { getSelectedDocuments } from "../../../samples/selectors";
import { analyze } from "../../actions";
import { IndexSelector } from "./IndexSelector";
import { SelectedSamples } from "./SelectedSamples";

const MultiSummary = ({ samples, selected }) => {
    const product = selected.length * samples.length;

    if (product === 0) {
        return null;
    }

    return (
        <div style={{ textAlign: "left" }}>
            {product} job
            {product === 1 ? "" : "s"} will be started
        </div>
    );
};

const getInitialState = ({ defaultSubtraction }) => ({
    algorithm: "pathoscope_bowtie",
    selected: [],
    subtraction: defaultSubtraction,
    error: ""
});

export class CreateAnalysis extends React.Component {
    constructor(props) {
        super(props);
        this.state = getInitialState(props);
    }

    handleEnter = () => {
        this.setState(getInitialState(this.props));
    };

    handleSelectAlgorithm = e => {
        this.setState({ algorithm: e.target.value });
    };

    handleSelectIndex = index => {
        this.setState({ selected: xorBy(this.state.selected, [index], "id"), error: "" });
    };

    handleSelectSubtraction = e => {
        this.setState({ subtraction: e.target.value });
    };

    handleSubmit = e => {
        e.preventDefault();

        if (!this.state.selected.length) {
            return this.setState({ error: "Please select reference(s)" });
        }

        this.props.onAnalyze(
            this.props.documents,
            this.state.selected,
            this.state.algorithm,
            this.state.subtraction,
            this.props.userId
        );

        this.props.onHide();
    };

    render() {
        const { selected, algorithm } = this.state;

        const show = !!(this.props.documents && this.props.documents.length);

        return (
            <Modal show={show} onHide={this.props.onHide} onEnter={this.handleEnter}>
                <Modal.Header>Analyze</Modal.Header>
                <form onSubmit={this.handleSubmit}>
                    <Modal.Body>
                        <SelectedSamples samples={this.props.documents} />
                        <AlgorithmSelect
                            value={algorithm}
                            onChange={this.handleSelectAlgorithm}
                            hasHmm={this.props.hasHmm}
                        />
                        <IndexSelector
                            indexes={this.props.indexes}
                            onSelect={this.handleSelectIndex}
                            selected={selected}
                            error={this.state.error}
                        />
                        <Subtrac
                    </Modal.Body>
                    <Modal.Footer>
                        <Flex alignItems="center">
                            <FlexItem grow={1}>
                                <MultiSummary
                                    algorithm={algorithm}
                                    samples={this.props.documents}
                                    selected={this.state.selected}
                                />
                            </FlexItem>
                            <FlexItem>
                                <Button type="submit" bsStyle="primary" icon="play" disabled={!!this.state.error}>
                                    Start
                                </Button>
                            </FlexItem>
                        </Flex>
                    </Modal.Footer>
                </form>
            </Modal>
        );
    }
}

const mapStateToProps = state => ({
    documents: getSelectedDocuments(state),
    hasHmm: !!state.hmms.total_count,
    indexes: state.analyses.readyIndexes,
    defaultSubtraction: state.samples.detail.subtraction.id,
    userId: state.account.id
});

const mapDispatchToProps = dispatch => ({
    onAnalyze: (samples, references, algorithm, subtractionId, userId) => {
        forEach(samples, ({ id }) => {
            forEach(references, ({ refId }) => {
                dispatch(analyze(id, refId, algorithm, subtractionId, userId));
            });
        });
    },
    onHide: () => {
        dispatch(pushState({}));
    }
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(CreateAnalysis);
