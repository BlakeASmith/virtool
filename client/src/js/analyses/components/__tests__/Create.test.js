import { CreateAnalysis } from "../Create";

describe("<CreateAnalysis />", () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
            show: true,
            samples: null,
            id: "123abc",
            hasHmm: false,
            refIndexes: [],
            selected: [],
            userId: "test-user",
            onSubmit: jest.fn(),
            onHide: jest.fn(),
            onAnalyze: jest.fn()
        };
    });

    it("renders correctly", () => {
        wrapper = shallow(<CreateAnalysis {...props} />);
        expect(wrapper).toMatchSnapshot();

        wrapper.setState({ selected: [{ id: "test1" }, { id: "test2" }] });
        expect(wrapper.find({ style: { float: "left" } })).toMatchSnapshot();

        wrapper.setProps({ samples: ["test", "example"] });
        expect(wrapper.find({ style: { float: "left" } })).toMatchSnapshot();

        wrapper.setProps({ samples: ["test"] });
        wrapper.setState({ selected: [{ id: "test" }] });
        expect(wrapper.find({ style: { float: "left" } })).toMatchSnapshot();
    });
});
