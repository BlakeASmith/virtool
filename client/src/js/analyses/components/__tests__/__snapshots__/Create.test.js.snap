// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateAnalysis /> should render 1`] = `
<ModalDialog
  label="create"
  onEnter={[Function]}
  onHide={[MockFunction]}
  show={true}
  size="lg"
>
  <Modal__DialogHeader>
    Analyze
  </Modal__DialogHeader>
  <form
    onSubmit={[Function]}
  >
    <Modal__DialogBody>
      <SelectedSamples
        samples={
          Array [
            Object {
              "id": "foo",
              "name": "foo",
            },
            Object {
              "id": "bar",
              "name": "Bar",
            },
          ]
        }
      />
      <AlgorithmSelect
        onChange={[Function]}
        value="pathoscope_bowtie"
      />
      <SubtractionSelector
        onChange={[Function]}
      />
      <IndexSelector
        error=""
        indexes={
          Array [
            Object {
              "id": "ref1",
              "reference": Object {
                "id": "fer1",
                "name": "Ref1",
              },
              "version": 3,
            },
            Object {
              "id": "ref2",
              "reference": Object {
                "id": "fer2",
                "name": "Ref2",
              },
              "version": 1,
            },
          ]
        }
        onSelect={[Function]}
        selected={Array []}
      />
    </Modal__DialogBody>
    <Modal__DialogFooter>
      <Flex
        alignContent="stretch"
        alignItems="center"
        direction="row"
        justifyContent="flex-start"
        wrap="nowrap"
      >
        <FlexItem
          alignSelf={null}
          basis="auto"
          grow={1}
          shrink={1}
        >
          <MultiSummary
            algorithm="pathoscope_bowtie"
            samples={
              Array [
                Object {
                  "id": "foo",
                  "name": "foo",
                },
                Object {
                  "id": "bar",
                  "name": "Bar",
                },
              ]
            }
            selected={Array []}
          />
        </FlexItem>
        <FlexItem
          alignSelf={null}
          basis="auto"
          grow={0}
          shrink={1}
        >
          <Button
            bsStyle="primary"
            disabled={false}
            icon="play"
            pullRight={false}
            tipPlacement="top"
            type="submit"
          >
            Start
          </Button>
        </FlexItem>
      </Flex>
    </Modal__DialogFooter>
  </form>
</ModalDialog>
`;

exports[`<CreateAnalysis /> should render with algorithm selected 1`] = `
<ModalDialog
  label="create"
  onEnter={[Function]}
  onHide={[MockFunction]}
  show={true}
  size="lg"
>
  <Modal__DialogHeader>
    Analyze
  </Modal__DialogHeader>
  <form
    onSubmit={[Function]}
  >
    <Modal__DialogBody>
      <SelectedSamples
        samples={
          Array [
            Object {
              "id": "foo",
              "name": "foo",
            },
            Object {
              "id": "bar",
              "name": "Bar",
            },
          ]
        }
      />
      <AlgorithmSelect
        onChange={[Function]}
        value="nuvs"
      />
      <SubtractionSelector
        onChange={[Function]}
      />
      <IndexSelector
        error=""
        indexes={
          Array [
            Object {
              "id": "ref1",
              "reference": Object {
                "id": "fer1",
                "name": "Ref1",
              },
              "version": 3,
            },
            Object {
              "id": "ref2",
              "reference": Object {
                "id": "fer2",
                "name": "Ref2",
              },
              "version": 1,
            },
          ]
        }
        onSelect={[Function]}
        selected={Array []}
      />
    </Modal__DialogBody>
    <Modal__DialogFooter>
      <Flex
        alignContent="stretch"
        alignItems="center"
        direction="row"
        justifyContent="flex-start"
        wrap="nowrap"
      >
        <FlexItem
          alignSelf={null}
          basis="auto"
          grow={1}
          shrink={1}
        >
          <MultiSummary
            algorithm="nuvs"
            samples={
              Array [
                Object {
                  "id": "foo",
                  "name": "foo",
                },
                Object {
                  "id": "bar",
                  "name": "Bar",
                },
              ]
            }
            selected={Array []}
          />
        </FlexItem>
        <FlexItem
          alignSelf={null}
          basis="auto"
          grow={0}
          shrink={1}
        >
          <Button
            bsStyle="primary"
            disabled={false}
            icon="play"
            pullRight={false}
            tipPlacement="top"
            type="submit"
          >
            Start
          </Button>
        </FlexItem>
      </Flex>
    </Modal__DialogFooter>
  </form>
</ModalDialog>
`;

exports[`<CreateAnalysis /> should render with error 1`] = `
<ModalDialog
  label="create"
  onEnter={[Function]}
  onHide={[MockFunction]}
  show={true}
  size="lg"
>
  <Modal__DialogHeader>
    Analyze
  </Modal__DialogHeader>
  <form
    onSubmit={[Function]}
  >
    <Modal__DialogBody>
      <SelectedSamples
        samples={
          Array [
            Object {
              "id": "foo",
              "name": "foo",
            },
            Object {
              "id": "bar",
              "name": "Bar",
            },
          ]
        }
      />
      <AlgorithmSelect
        onChange={[Function]}
        value="pathoscope_bowtie"
      />
      <SubtractionSelector
        onChange={[Function]}
      />
      <IndexSelector
        error="Please select reference(s)"
        indexes={
          Array [
            Object {
              "id": "ref1",
              "reference": Object {
                "id": "fer1",
                "name": "Ref1",
              },
              "version": 3,
            },
            Object {
              "id": "ref2",
              "reference": Object {
                "id": "fer2",
                "name": "Ref2",
              },
              "version": 1,
            },
          ]
        }
        onSelect={[Function]}
        selected={Array []}
      />
    </Modal__DialogBody>
    <Modal__DialogFooter>
      <Flex
        alignContent="stretch"
        alignItems="center"
        direction="row"
        justifyContent="flex-start"
        wrap="nowrap"
      >
        <FlexItem
          alignSelf={null}
          basis="auto"
          grow={1}
          shrink={1}
        >
          <MultiSummary
            algorithm="pathoscope_bowtie"
            samples={
              Array [
                Object {
                  "id": "foo",
                  "name": "foo",
                },
                Object {
                  "id": "bar",
                  "name": "Bar",
                },
              ]
            }
            selected={Array []}
          />
        </FlexItem>
        <FlexItem
          alignSelf={null}
          basis="auto"
          grow={0}
          shrink={1}
        >
          <Button
            bsStyle="primary"
            disabled={true}
            icon="play"
            pullRight={false}
            tipPlacement="top"
            type="submit"
          >
            Start
          </Button>
        </FlexItem>
      </Flex>
    </Modal__DialogFooter>
  </form>
</ModalDialog>
`;

exports[`<CreateAnalysis /> should render with index selected 1`] = `
<ModalDialog
  label="create"
  onEnter={[Function]}
  onHide={[MockFunction]}
  show={true}
  size="lg"
>
  <Modal__DialogHeader>
    Analyze
  </Modal__DialogHeader>
  <form
    onSubmit={[Function]}
  >
    <Modal__DialogBody>
      <SelectedSamples
        samples={
          Array [
            Object {
              "id": "foo",
              "name": "foo",
            },
            Object {
              "id": "bar",
              "name": "Bar",
            },
          ]
        }
      />
      <AlgorithmSelect
        onChange={[Function]}
        value="pathoscope_bowtie"
      />
      <SubtractionSelector
        onChange={[Function]}
      />
      <IndexSelector
        error=""
        indexes={
          Array [
            Object {
              "id": "ref1",
              "reference": Object {
                "id": "fer1",
                "name": "Ref1",
              },
              "version": 3,
            },
            Object {
              "id": "ref2",
              "reference": Object {
                "id": "fer2",
                "name": "Ref2",
              },
              "version": 1,
            },
          ]
        }
        onSelect={[Function]}
        selected={
          Array [
            Object {
              "id": "ref1",
              "refId": "fer1",
            },
          ]
        }
      />
    </Modal__DialogBody>
    <Modal__DialogFooter>
      <Flex
        alignContent="stretch"
        alignItems="center"
        direction="row"
        justifyContent="flex-start"
        wrap="nowrap"
      >
        <FlexItem
          alignSelf={null}
          basis="auto"
          grow={1}
          shrink={1}
        >
          <MultiSummary
            algorithm="pathoscope_bowtie"
            samples={
              Array [
                Object {
                  "id": "foo",
                  "name": "foo",
                },
                Object {
                  "id": "bar",
                  "name": "Bar",
                },
              ]
            }
            selected={
              Array [
                Object {
                  "id": "ref1",
                  "refId": "fer1",
                },
              ]
            }
          />
        </FlexItem>
        <FlexItem
          alignSelf={null}
          basis="auto"
          grow={0}
          shrink={1}
        >
          <Button
            bsStyle="primary"
            disabled={false}
            icon="play"
            pullRight={false}
            tipPlacement="top"
            type="submit"
          >
            Start
          </Button>
        </FlexItem>
      </Flex>
    </Modal__DialogFooter>
  </form>
</ModalDialog>
`;

exports[`<CreateAnalysis /> should render with summary 1`] = `
<ModalDialog
  label="create"
  onEnter={[Function]}
  onHide={[MockFunction]}
  show={true}
  size="lg"
>
  <Modal__DialogHeader>
    Analyze
  </Modal__DialogHeader>
  <form
    onSubmit={[Function]}
  >
    <Modal__DialogBody>
      <SelectedSamples
        samples={
          Array [
            Object {
              "id": "foo",
              "name": "foo",
            },
            Object {
              "id": "bar",
              "name": "Bar",
            },
          ]
        }
      />
      <AlgorithmSelect
        onChange={[Function]}
        value="nuvs"
      />
      <SubtractionSelector
        onChange={[Function]}
      />
      <IndexSelector
        error=""
        indexes={
          Array [
            Object {
              "id": "ref1",
              "reference": Object {
                "id": "fer1",
                "name": "Ref1",
              },
              "version": 3,
            },
            Object {
              "id": "ref2",
              "reference": Object {
                "id": "fer2",
                "name": "Ref2",
              },
              "version": 1,
            },
          ]
        }
        onSelect={[Function]}
        selected={
          Array [
            Object {
              "id": "ref1",
              "refId": "fer1",
            },
          ]
        }
      />
    </Modal__DialogBody>
    <Modal__DialogFooter>
      <Flex
        alignContent="stretch"
        alignItems="center"
        direction="row"
        justifyContent="flex-start"
        wrap="nowrap"
      >
        <FlexItem
          alignSelf={null}
          basis="auto"
          grow={1}
          shrink={1}
        >
          <MultiSummary
            algorithm="nuvs"
            samples={
              Array [
                Object {
                  "id": "foo",
                  "name": "foo",
                },
                Object {
                  "id": "bar",
                  "name": "Bar",
                },
              ]
            }
            selected={
              Array [
                Object {
                  "id": "ref1",
                  "refId": "fer1",
                },
              ]
            }
          />
        </FlexItem>
        <FlexItem
          alignSelf={null}
          basis="auto"
          grow={0}
          shrink={1}
        >
          <Button
            bsStyle="primary"
            disabled={false}
            icon="play"
            pullRight={false}
            tipPlacement="top"
            type="submit"
          >
            Start
          </Button>
        </FlexItem>
      </Flex>
    </Modal__DialogFooter>
  </form>
</ModalDialog>
`;
